sameinad.py

```diff
@@ -0,0 +1,401 @@
+#----------------------------------------------------------------------------------------------------
+# Setup and Environment Configuration
+#----------------------------------------------------------------------------------------------------
+
+from sqlalchemy import create_engine
+from shiny import App, render, ui
+import pandas as pd
+import geopandas as gpd
+from shapely import wkt
+import folium
+from folium.plugins import MarkerCluster
+from matplotlib import cm, colors
+import numpy as np
+import plotly.express as px
+from shinywidgets import output_widget, render_widget
+import os
+from dotenv import load_dotenv
+
+# Load environment variables from the .env file
+load_dotenv()
+
+# Fetch database credentials from environment variables
+db_user = os.getenv("DB_USER")
+db_password = os.getenv("DB_PASSWORD")
+db_host = os.getenv("DB_HOST")
+db_port = os.getenv("DB_PORT")
+db_name = os.getenv("DB_NAME")
+
+# Check if all variables are set
+if not all([db_user, db_password, db_host, db_port, db_name]):
+    raise ValueError("One or more environment variables are not set.")
+
+# Create the database engine
+engine = create_engine(f"postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}")
+
+
+#----------------------------------------------------------------------------------------------------
+# Data Fetching and Processing
+#----------------------------------------------------------------------------------------------------
+
+# Function to fetch houses and kingdoms data with explicit mapping
+def fetch_houses_and_kingdoms():
+    query = """
+    SELECT 
+        h.name AS house_name,
```

Comment:
```suggestion
        regexp_replace(h.name, '^House ','') AS "House Name",
```
og sleppa þá regexinu fyrir neðan og ef þið viljið hafa bil í dálka heiti þá gerið þið með tvöföldum gæsalöppum